mcf-browser/src/App.tsx

```diff
@@ -189,60 +196,73 @@ class App extends Component<{}, AppStateType> {
         files: [...prevState.files, res.data],
       }));
     }
-    this.submitFileList(newFiles);
+    await this.submitFileList(newFiles);
   }
 
   /**
-   * Saves the list of new blob files to App' state list of files and submits
-   * them to be parsed.
-   * @param {Array<Blob>} fileList File blobs selcted by user via file picker.
-   */
+  * Saves the list of new blob files to App' state list of files and submits
+  * them to be parsed.
+  * @param {Array<Blob>} fileList File blobs selcted by user via file picker.
+  */
   async uploadFiles(fileList: Blob[]) {
     for (const file of fileList) {
       await this.setState((prevState) => ({
         files: [...prevState.files, file],
       }));
     }
-    this.submitFileList(this.state.files);
+    await this.submitFileList(this.state.files);
   }
 
   /**
-   * Passes list of file blobs to the 'back-end' API to be parsed and loaded
-   * into memory.
-   * @param {Array<Blob>} fileList File blobs to be parsed by back-end.
-   */
+  * Passes list of file blobs to the 'back-end' API to be parsed and loaded
+  * into memory.
+  * @param {Array<Blob>} fileList File blobs to be parsed by back-end.
+  */
   submitFileList(fileList: Blob[]) {
     this.setState({loading: true});
 
-    API.readFileList(fileList).then((res: any) => {
+    const res = API.readFileList(fileList);
+    res.then((res: API.ReadFileListResponse) => {
       this.setState(() => ({
         parsingErrs: res['errMsgs'],
         subjNodes: res['localNodes'],
+        timeData: res['timeData'],
         loading: false,
       }), () => this.handleHashChange());
     });
+    res.catch(
+        () => this.setState((prevState) => {
```

Comment:
you should be able to just chain the catch like:
```
res.then(() => {}).catch(() => {});
```
and for setting state, I believe you can just do `this.setState({loading: false});`

Reply:
It didn't let me do that before because the `function` in `.then(function)` doesn't return a Promise. Should I return a empty Promise or should I just keep it as is?

Reply:
oh, interesting, ok, can keep it as is!